lib/sycamore/sycamore/writer.py

```diff
@@ -97,6 +98,14 @@ def opensearch(
         from typing import Any
         import copy
 
+        if os_client_args is None:
+            os_client_args = self.context.config.get(Config.OPENSEARCH_CLIENT_CONFIG)
+        assert os_client_args is not None
+
+        if not index_name:
```

Comment:
See comment above, I tend to think this should be passed in on each call

Reply:
replied above